### ISSUE TITLE
specialize() and compute_at() interaction failing

### ISSUE THREAD

**Issue by shoaibkamil** [halide/Halide#3926]
Given this code:
```
#include "Halide.h"

using namespace Halide;

int main(int argc, char* argv[]) {
  Func f("f"), g("g");
  Var x("x"), y("y");
  Var tx("tx"), ty("ty");
  Param<bool> param;

  f(x) = x;
  g(x, y) = f(x) + select(param, 1, 2);

#if 0
  g.gpu_tile(x, y, tx, ty, 8, 8, TailStrategy::GuardWithIf);
#else
  g.specialize(param).gpu_tile(x, y, tx, ty, 8, 8, TailStrategy::GuardWithIf);
  g.specialize(!param).gpu_tile(x, y, tx, ty, 8, 8, TailStrategy::GuardWithIf);
  g.specialize_fail("Unknown");
#endif
  f.in().compute_at(g, tx);

  g.compile_to_file("foo", {param});

}
```

This gives an error:
```
Error:
Func "f_global_wrapper" is computed at the following invalid location:
  f_global_wrapper.compute_at(g, tx);
Legal locations for this function are:
  f_global_wrapper.compute_root();
  f_global_wrapper.compute_at(g, Var::outermost());
  f_global_wrapper.compute_at(g, y);
  f_global_wrapper.compute_at(g, x);
"f_global_wrapper" is used in the following places:
  for g.s0.y.y:
    for g.s0.x.x:
      for g.s0.y.ty:
        for g.s0.x.tx:
          g uses f_global_wrapper
  for g.s0.y.y:
    for g.s0.x.x:
      for g.s0.y.ty:
        for g.s0.x.tx:
          g uses f_global_wrapper
  for g.s0.y:
    for g.s0.x:
      g uses f_global_wrapper
```

However, the last use (causing the violation) is spurious, because the `specialize_fail()` ensures this branch will never be reached.  Looking at the IR, I see:
```
...
    } else if ((uint1)1) {
      assert((uint1)0, halide_error_specialize_fail("Unknown"))
    } else {
      // violating usage here
    }
....
```

**Comment by steven-johnson** [comment]
Is this resolved by #3927?

**Comment by shoaibkamil** [comment]
Yup, resolved.
